tensorflow_datasets/core/dataset_builder.py

```diff
@@ -218,6 +218,13 @@ def canonical_version(self):
     else:
       return self.VERSION
 
+  @utils.classproperty
+  @classmethod
+  @utils.memoize()
+  def code_dir(cls) -> str:
```

Comment:
After thinking more about this, it would be nice to rename this `code_path` and return a pathlib.Path.

It would allow to write `builder.code_path.name` or `builder.code_path.parent / 'checksums.txt'` which seems nicer.

I also think it should returns the python file, not the directory. Indeed, there are some few cases where getting the path will be helpful, like documenting the source code location, or for the refactoring script.